### PR TITLE
Alwasy enable ceph msgrv2 port & pass kernel mount options to rook-ceph operator

### PR DESCRIPTION
**Pull request by malayparida2000**
Jira-https://issues.redhat.com/browse/RHSTOR-2516

For easier enablement of new features like in-transit encryption &
compression we need the ceph msgr v2 port. So we will use the msgrv2
port instead of the old v1 port by default.

The cephFS kernel mount option is set as a key in the ocsoperator
configmap. The value of the key is passed to rook as a env variable.
The default is "ms_mode=prefer-crc" and if in transit encryption is
enabled, the value is set to "ms_mode=secure". The rook operator
pod is restarted if the value of the key is changed.

### CODE REVIEW

**Comment by malayparida2000** [comment]
This PR is now ready for review, The rook API changes, make deps-update changes & the rook latest image changes can be removed if they merge from other PRs. They are for now part of the PR for complete testing

File: controllers/storagecluster/cephcluster.go
Comment (travisn) [inline controllers/storagecluster/cephcluster.go:1212]: This configmap appears to be created/updated in multiple places, here and in ocsoperator_config.go. Will that not cause conflicts?

File: controllers/storagecluster/cephcluster.go
Comment (Madhu-1) [inline controllers/storagecluster/cephcluster.go:1233]: Not related to by t pass log and won't return anything from this function, just return an error and log it outside?

File: controllers/storagecluster/cephcluster.go
Comment (Madhu-1) [inline controllers/storagecluster/cephcluster.go:1220]: define a const for `CSI_CEPHFS_KERNEL_MOUNT_OPTIONS` and use it

File: controllers/storagecluster/cephcluster.go
Comment (Madhu-1) [inline controllers/storagecluster/cephcluster.go:1219]: instead of `restartNeeded` you can make use of the result returned from `CreateOrUpdate` itself.

File: controllers/storagecluster/cephcluster.go
Comment (Madhu-1) [inline controllers/storagecluster/cephcluster.go:1209]: As this is business logic You can move this to line 1230?

File: controllers/ocsinitialization/ocsoperator_config.go
Comment (Madhu-1) [inline controllers/ocsinitialization/ocsoperator_config.go:29]: Is this required to be set? Because `prefer-crc` is getting set in setCephFSKernelMountOption also.

**Comment by malayparida2000** [comment]
The flow for both the ocs-operator-config cm was too much cluttered & messed up. Have made another PR to refactor & streamline that flow. That PR has been approved but due to some temporary ci  issue, there is a delay in merging. I wanted to make the PR review ready so have kept the changes expected to be merged from that PR in a separate commit & the required changes are on top. Once  #1948 merges I will rebase this. Meanwhile, I need your reviews for the core changes on the top 4 commits.

File: controllers/storagecluster/cephcluster.go
Comment (malayparida2000) [reply controllers/storagecluster/cephcluster.go]: Going forward ocsinitializer will just create the configmap with the keys so that rook operator doesn't through errors. (as rook operator expects these value so if the configmap or the key doesn't exist it will stay in container create config error). So we need to create the configmap with some default value when the operator is installed & we can later update them with correct values in storagecluster controller.

File: controllers/storagecluster/cephcluster.go
Comment (malayparida2000) [reply controllers/storagecluster/cephcluster.go]: This is now addressed in another PR.

File: controllers/storagecluster/cephcluster.go
Comment (malayparida2000) [reply controllers/storagecluster/cephcluster.go]: Done

File: controllers/ocsinitialization/ocsoperator_config.go
Comment (malayparida2000) [reply controllers/ocsinitialization/ocsoperator_config.go]: Going forward ocsinitializer will just create the configmap with the keys so that rook operator doesn't through errors. (as rook operator expects these value so if the configmap or the key doesn't exist it will stay in container create config error). So we need to create the configmap with some default value when the operator is installed & we can later update them with correct values in storagecluster controller.

**Comment by travisn** [comment]
LGTM after @Madhu-1 also reviews

File: controllers/storagecluster/ocs_operator_config.go
Comment (Madhu-1) [inline controllers/storagecluster/ocs_operator_config.go:102]: Changes LGTM, I hope you have tested how this behaves when v1 ports are used in upgraded clusters.

Comment (malayparida2000) [reply controllers/storagecluster/ocs_operator_config.go:102]: Haven't been able to test the case with upgrades with Mon failover

Comment (malayparida2000) [reply controllers/storagecluster/ocs_operator_config.go:102]: Let me do it

**Review by iamniting — APPROVED** [review]
changes looks good to me. holding the changes. pls, remove the hold once it is tested.

/hold

**Comment by malayparida2000** [comment]
Tested the feature for both fresh install & upgrade case+mon failover.
Removing hold.
/unhold

**Pull request merged**
